go/app/main.go

```diff
@@ -33,24 +55,143 @@ func addItem(c echo.Context) error {
 	message := fmt.Sprintf("item received: %s", name)
 	res := Response{Message: message}
 
+	//ここからStep2
+	//jsonファイルの読み込み
+	raw, err := os.ReadFile("./items.json")
+	//エラーハンドリング
+	if err != nil {
+        fmt.Println("Read error")
+    }
+	//jsonに対応した構造体のインスタンスを生成
+	var itemsWrapper ItemsWrapper
+	//jsonをデコード
+	if err := json.Unmarshal(raw, &itemsWrapper); err != nil {
+		fmt.Println("Decode error")
+		fmt.Println(err)
+	}
+
+	//ここから画像
+	// フォームのファイルを取得
+	file, err := c.FormFile("image")
+	if err != nil {
+		fmt.Println("Image Get error")
+	}
+	//ファイルを開いて内容をsrcに代入
+	src, err := file.Open()
+	if err != nil {
+		fmt.Println("Image Open error")
+	}
+	defer src.Close()
+
+	fileModel := strings.Split(file.Filename, ".")
+	fileName := fileModel[0]
+	extension := fileModel[1]
+	fileName_hash := sha_256([]byte(fileName))
+
+	// 保存用のディレクトリを作成する（存在していなければ、保存用のディレクトリを新規作成）
+	//err = os.MkdirAll("./images", os.ModePerm)
+	//if err != nil {
+	//	fmt.Println("Mkdir error")
+	//}
+
+	// 保存用のファイルを作成する
+	dst, err := os.Create("../images/"+fmt.Sprintf("%s.%s",string(fileName_hash),extension))
+	if err != nil {
+		fmt.Println("Mkfile error")
+		fmt.Println(err)
+	}
+	defer dst.Close()
+
+	// アップロードされたファイルの内容を保存用のファイルにコピーする
+	_, err = io.Copy(dst, src)
+	if err != nil {
+		fmt.Println("Copyfile error")
+	}
+	fmt.Println("アップロード成功!")
+	// ここまで画像
+	var imageName string = fileName_hash + "." + extension
```

Comment:
[nits] goで直接文字列を宣言する場合はこちらの記法を使うことの方が多いです。(好みの問題ですが)

```suggestion
	imageName := fileName_hash + "." + extension
```